extension-style/src/main/java/com/mapbox/maps/extension/style/projection/generated/ProjectionExt.kt

```diff
@@ -0,0 +1,66 @@
+// This file is generated.
+@file:JvmName("StyleProjectionUtils")
+
+package com.mapbox.maps.extension.style.projection.generated
+
+import com.mapbox.maps.StylePropertyValueKind
+import com.mapbox.maps.extension.style.StyleInterface
+import com.mapbox.maps.extension.style.layers.properties.generated.ProjectionName
+import com.mapbox.maps.extension.style.utils.unwrapToAny
+
+/**
+ * Extension function to get the projection name provided by the Style Extension.
+ *
+ * @return projection [ProjectionName] that is currently applied to the map
+ */
+fun StyleInterface.getProjectionName(): ProjectionName {
+  getStyleProjectionProperty("name").apply {
+    return if (kind == StylePropertyValueKind.UNDEFINED) {
+      ProjectionName.MERCATOR
+    } else {
+      val projectionAsString = value.unwrapToAny() as String
+      return ProjectionName.valueOf(projectionAsString.uppercase())
+    }
+  }
+}
```

Comment:
We could probably just have getProjection/setProjection methods? the API should already be easy to use with these 2 extensions. Otherwise, we need to introduce additional project properties extension functions as well, which seems a bit overkill.

Reply:
Still wonder the necessity to expose `getProjectionName` and `setProjectionName` extension functions. Have we done similar extension in for example `terrain` and `light`?  I think it's a bit inconsistent with other APIs, here we are providing the property below the `Projection` abstraction.